### ISSUE TITLE
Have prediction tables

### ISSUE THREAD

**Issue by fily1212** [CRIPAC-DIG/A-PGNN#3]
Hello, 
I'm trying to integrate this algorithm in a evaluation framework of session-aware RS. 
I need to create a method that, given a user and a session with previous items (in test_set), this method return the probability table of each "next" item. 
Example: user 1, item: 1, 2, 4. return: 
items - probability
0 - 0.02
1 - 0.05
2- 0.05
3 - 0.001
4 - 0.2
5 - 0.3
.... (for each item) 
Or at least the list of the most probable items. 

is there a way to generate this from your code, and where? 
I think maybe I have to feed the method model.forward, but I can't understand in which way. Can you confirm me if I'm looking at the right method? 
thank you so much. 
Best,
Filippo

**Comment by ZM7** [comment]
Thanks for your attention. The 155th line of model_last.py,  "logits = tf.matmul(y1, b, transpose_b=True)", might meet your needs.

**Comment by ZM7** [comment]
> Hello,
> I'm trying to integrate this algorithm in a evaluation framework of session-aware RS.
> I need to create a method that, given a user and a session with previous items (in test_set), this method return the probability table of each "next" item.
> Example: user 1, item: 1, 2, 4. return:
> items - probability
> 0 - 0.02
> 1 - 0.05
> 2- 0.05
> 3 - 0.001
> 4 - 0.2
> 5 - 0.3
> .... (for each item)
> Or at least the list of the most probable items.
> 
> is there a way to generate this from your code, and where?
> I think maybe I have to feed the method model.forward, but I can't understand in which way. Can you confirm me if I'm looking at the right method?
> thank you so much.
> Best,
> Filippo

Thanks for your attention. The 155th line of model_last.py, "logits = tf.matmul(y1, b, transpose_b=True)", might meet your needs.
